webui-src/app/files/files_manager.js

```diff
@@ -0,0 +1,268 @@
+              return m('tr', [
+                m(
+                  'td',
+                  m('input[type=text]', {
+                    value: filename,
+                    disabled: isEditDisabled,
+                    oninput: (e) => {
+                      sharedDirArr[index].filename = e.target.value;
+                    },
+                  })
+                ),
+                m(
+                  'td',
+                  m('input[type=text]', {
+                    value: virtualname,
+                    disabled: isEditDisabled,
+                    oninput: (e) => {
+                      sharedDirArr[index].virtualname = e.target.value;
+                    },
+                  })
+                ),
+                m(
+                  'td.share-flags',
+                  Object.keys(sharedFlags).map((flag) => {
+                    return [
+                      m(`input.share-flags-check[type=checkbox][id=${flag}]`, {
+                        checked: sharedFlags[flag],
+                        disabled: isEditDisabled,
+                      }),
+                      m(
+                        `label.share-flags-label[for=${flag}]`,
+                        {
+                          onclick: () => {
+                            if (isEditDisabled) return;
+                            sharedFlags[flag] = !sharedFlags[flag];
+                            sharedDirArr[index].shareflags = futil.calcShareFlagsValue(sharedFlags);
+                          },
+                          style: isEditDisabled && { color: '#7D7D7D' },
+                        },
+                        m(
+                          // check the flag type then if its value is true then only render the icon
+                          flag === 'isAnonymousSearch'
+                            ? sharedFlags[flag]
+                              ? 'i.fas.fa-search'
+                              : 'span'
+                            : flag === 'isAnonymousDownload'
+                            ? sharedFlags[flag]
+                              ? 'i.fas.fa-download'
+                              : 'span'
+                            : sharedFlags[flag]
+                            ? 'i.fas.fa-eye'
+                            : 'span'
+                        )
+                      ),
+                    ];
+                  })
+                ),
+                m(
+                  'td',
+                  {
+                    // since this is not an input element, manually change color
+                    style: { color: isEditDisabled ? '#6D6D6D' : 'black' },
+                    onclick: () =>
+                      !isEditDisabled && widget.popupMessage(m(ManageVisibility, { parentGroups })),
+                  },
+                  parentGroups.length === 0
+                    ? 'All Friend nodes'
+                    : parentGroups.map((groupFlag) => `${futil.RsNodeGroupId[groupFlag]},`)
```

Comment:
nit: you could do a join so `,` at the end isn't added
```suggestion
                    : parentGroups.map((groupFlag) => futil.RsNodeGroupId[groupFlag]).join(', ')
```

Reply:
Yup, that's much better. Thanks : )